plugins/modules/azure_rm_eventhub.py

```diff
@@ -0,0 +1,434 @@
+            - Unknown
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+        required: False
+        type: str
+    sku:
+        description:
+            - The name of the SKU.
+            - Please see L(https://azure.microsoft.com/en-in/pricing/details/event-hubs/).
+        default: Basic
+        choices:
+            - Basic
+            - Standard
+        type: str
+    state:
+      description:
+          - Assert the state of the Event Hub.
+          - Use C(present) to create or update an event hub and C(absent) to delete it.
+      default: present
+      choices:
+          - absent
+          - present
+      type: str
+
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+
+author:
+    - Praveen Ghuge (@praveenghuge)
+    - Karl Dasan (@karldas30)
+    - Saurabh Malpani(@saurabh3796)
+'''
+EXAMPLES = '''
+
+- name: "Create Event Hub"
+  azure_rm_eventhub:
+    resource_group: testgroupans
+    location: eastus
+    namespace_name: myNamespace
+    name: myhub
+    tags:
+       - a: b
+    sku: free
+
+- name: Delete Event Hub
+  azure_rm_eventhub:
+    resource_group: testgroupans
+    name: myNamespace
+    state: absent
+
+- name: "Create Event Hub Namespace"
+  azure_rm_eventhub:
+    resource_group: testgroupans
+    location: eastus
+    namespace_name: myNamespace
+    tags:
+      a: b
+    sku: free
+
+- name: Delete Event Hub Namespace
+  azure_rm_eventhub:
+    resource_group: testgroupans
+    namespace_name: myNamespace
+    state: absent
+
+'''
+
+RETURN = '''
+state:
+    description:
+        - Current state of the Event Hub namesapce or Event Hub.
+    returned: always
+    type: dict
+    sample: {
+        "additional_properties": {"location": "East US"},
+        "critical": false,
+        "enabled": true,
+        "metric_id": null,
+        "name": "testnaedd3d22d3w",
+        "namespace_type": "eventHub",
+        "status": "Active",
+        "region": null,
+        "scale_unit": null,
+        "service_bus_endpoint": "https://testnaedd3d22d3w.servicebus.windows.net:443/",
+        "sku": "Basic",
+        "tags": {
+            "a": "b"
+        },
+        "message_retention_in_days": 7,
+        "partition_count": 4,
+        "partition_ids": ["0", "1", "2", "3"],
+        "updated_at": "2021-04-29T10:05:24.000Z",
+        "created_at": "2021-04-29T10:05:20.377Z",
+        "type": "Microsoft.eventHubs/namespaces"
+    }
+
+'''
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+    from azure.mgmt.eventhub.models import Eventhub, EHNamespace
+    from azure.mgmt.eventhub.models.sku import Sku
+    import time
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+partition_count_list = [i for i in range(1, 33)]
```

Comment:
According to Ansible Sanity definition specification should be a list of 1~32! Thank you very much!
```suggestion
partition_count_list = [1,2,3,4,5,6,7,8,......,31,32]
```